### PR TITLE
Add feature for error handler and core updates

### LINKED ISSUE #20: Error handling
Hi, @DIOHz0r 

I will put here all possible errors for the library.

### PR DESCRIPTION
**Pull request by DIOHz0r**
## Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: [Conventional Commit](http://conventionalcommits.org/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Update client code

Closes #20 

## What is the new behavior?
Adds error handler message for client

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Glpi/Api/Rest/ItemHandler.php [hunk 1]
```diff
@@ -249,4 +251,55 @@ public function deleteItem($itemType, $id, array $inputValues = [], array $query
...
+    *    Consumableitems => glpi_consumableitems,
+    *
+    * How to use this?. You can check the file 'example/magicCallExample.php' for reference.
+    *
+    * @param string $name of the itemtype
+    * @param mixed $arguments first is method, second is the input, and third is the query string
+    * @return array|null
+    * @throws InsufficientArgumentsException|BadEndpointException|InvalidArgumentException
+    */
+   public function __call($name, $arguments) {
```
Comment (btry) [inline src/Glpi/Api/Rest/ItemHandler.php:608]: As you use magic method here, you should add php annotation @ method to help IDE to know **create** **read** **update** and **delete**

This is convenient for maintainers iof the library, but also the developers using it. Exampes are in this diff, where ou deleted these a nnotations.

File: src/Glpi/Api/Rest/ItemHandler.php [hunk 2]
```diff
@@ -249,4 +251,55 @@ public function deleteItem($itemType, $id, array $inputValues = [], array $query
...
+      $method = strtolower($arguments[0]);
+      $input = isset($arguments[1]) ? $arguments[1] : [];
+      $params = isset($arguments[2]) ? $arguments[2] : [];
+      $result = null;
+      switch ($method) {
+         case 'create':
+            $result = $this->addItem($name, $input);
+            break;
+         case 'read':
+            $result = $this->getAnItem($name, (int)$input, $params);
```
Comment (btry) [inline src/Glpi/Api/Rest/ItemHandler.php]: getItem() would be closer to the 3 other methods. I think this would be easier to use.

**Review by btry — CHANGES_REQUESTED** [review]

**Comment by btry** [comment]
This is not obvious in this PR, but I guess there is a magic method to handle $itemHandler->SomeItemtype()

This is what I wanted, nice. If you add @ method annotation to declare to an IDE all itemtypes of GLPI this will  be convenient for developers. A plugin which adds new itemtypes only would have to create a composer package declaring extra @ method annotation to extend the intelligence and knowledge of the IDE with extra endpoints. Going further, maybe the @ method annotations should all move to a dedicated package, like atoum does.

What do you think about this ?

**Comment by btry** [comment]
Also I don't like this PR being labelled **bug** because you implement features, not bugfixes :D

File: src/Glpi/Api/Rest/ItemHandler.php (hunk 2 above)
Comment (DIOHz0r) [reply src/Glpi/Api/Rest/ItemHandler.php]: Do you mean to change the name of getAnItem method to getItem, don't you?

Comment (btry) [reply src/Glpi/Api/Rest/ItemHandler.php]: Yes

File: src/Glpi/Api/Rest/Client.php [hunk 3]
```diff
@@ -47,8 +47,8 @@ class Client {
    /** @var string Session token obtained after initSession() */
    private $sessionToken = null;
 
-   public function __construct($url) {
-      $this->httpClient = new HttpClient();
+   public function __construct($url, HttpClient $httpClient) {
```
Comment (btry) [inline src/Glpi/Api/Rest/Client.php]: Interface is your friend ;)

File: src/Glpi/Api/Rest/ItemHandler.php (hunk 1 above)
Comment (DIOHz0r) [reply src/Glpi/Api/Rest/ItemHandler.php:608]: done in commit f7dc8fb70b

**Review by ajsb85 — APPROVED** [review]
Reviewed

**Pull request merged**